File: 70/Solution_70.cpp
#include <iostream>
#include <vector>

int climbStairs(int n){
    if (n <= 2){
        return n;
    }

    std::vector<int> dp(n);

    dp[0] = 0;
    dp[1] = 1;
    dp[2] = 2;

    for (int i = 3 ; i <= n ; i++){
        dp[i] = dp[i-1] + dp[i-2];
    }

    return dp[n];
}

int main(){
    std::cout << climbStairs(5) << std::endl;
    return 0;
}

File: 70/Solution_70.py
class Solution_70:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        p, q = 1, 2
        for _ in range(n - 2):
            r = p + q
            p, q = q, r
        return q  # or return r (both work)
